From d6227d0b7f4336a876e80371683dee9a7588991d Mon Sep 17 00:00:00 2001
From: Shane Volpe <shanevolpe@gmail.com>
Date: Sat, 6 Oct 2012 11:26:16 -0700
Subject: [PATCH 1/8] Initial commit

---
 README.md | 4 ++++
 1 file changed, 4 insertions(+)
 create mode 100644 README.md

diff --git a/README.md b/README.md
new file mode 100644
index 0000000..fbf4e6e
--- /dev/null
+++ b/README.md
@@ -0,0 +1,4 @@
+pyside-oe-layer
+===============
+
+Pyside layer for OpenEmbedded
\ No newline at end of file
-- 
1.8.5.2 (Apple Git-48)

