diff -Naur pyside-qt4.7+1.0.2-orig/CMakeLists.txt pyside-qt4.7+1.0.2/CMakeLists.txt
--- git/CMakeLists.txt	2011-04-28 22:43:01.000000000 +0200
+++ git/CMakeLists.txt	2011-05-21 22:49:28.043397002 +0200
@@ -89,6 +89,7 @@
 set(ENABLE_MAC "0")
 set(ENABLE_WIN "0")
 set(ENABLE_SIMULATOR "0")
+set(ENABLE_QWS "0")
 if(Q_WS_X11)
     set(ENABLE_X11 "1")
     if(Q_WS_MAEMO_5)
@@ -105,6 +106,9 @@
 elseif(Q_WS_SIMULATOR)
     set(ENABLE_SIMULATOR "1")
     set(AUTO_OS "simulator")
+elseif(1)
+    set(ENABLE_QWS "1")
+    set(AUTO_OS "qws")
 else()
     message(FATAL_ERROR "OS not supported")
 endif()
diff --git a/PySide/global.h.in b/PySide/global.h.in
index c85ba04..98bc117 100644
--- /dev/null	2012-06-06 10:06:23.975309641 -0400
+++ git/PySide/global.h.in	2012-10-08 14:53:47.952118932 -0400
@@ -343,6 +343,8 @@ QT_END_HEADER
   #include "@QT_QTGUI_INCLUDE_DIR@/QX11EmbedWidget"
 #elif @ENABLE_MAC@
   #include "@QT_QTGUI_INCLUDE_DIR@/qmacstyle_mac.h"
+#elif @ENABLE_QWS@
+  #include "@QT_QTGUI_INCLUDE_DIR@/QWSServer"
 #endif
 #include "@QT_QTXML_INCLUDE_DIR@/QtXml"
 #include "@QT_QTUITOOLS_INCLUDE_DIR@/QtUiTools"
--- /dev/null	2012-06-06 10:06:23.975309641 -0400
+++ git/PySide/QtCore/typesystem_core_qws.xml	2012-10-08 14:53:47.952118932 -0400
@@ -0,0 +1,26 @@
+<?xml version="1.0"?>
+<!--
+    This file is part of PySide project.
+    Copyright (C) 2011 Nokia Corporation and/or its subsidiary(-ies).
+    Contact: PySide team <contact@pyside.org>
+
+    This library is free software; you can redistribute it and/or
+    modify it under the terms of the GNU Lesser General Public
+    License as published by the Free Software Foundation; either
+    version 2.1 of the License, or (at your option) any later version.
+
+    This library is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+    Lesser General Public License for more details.
+
+    You should have received a copy of the GNU Lesser General Public
+    License along with this library; if not, write to the Free Software
+    Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+-->
+<typesystem package="PySide.QtCore">
+  <primitive-type name="Qt::HANDLE" target-lang-api-name="PyLong">
+    <!-- FIXME APIExtractor or shiboken do not support multiple includes by primitive type -->
+    <include file-name="QTextDocument" location="global"/>
+  </primitive-type>
+</typesystem>
--- git/PySide/QtGui/CMakeLists.txt b/PySide/QtGui/CMakeLists.txt 2012-06-06 10:06:23.975309641 -0400
+++ git/PySide/QtGui/CMakeLists.txt b/PySide/QtGui/CMakeLists.txt 2014-11-06 08:04:10.125504632 -0400
@@ -13,6 +13,14 @@ if(ENABLE_X11)
             ${CMAKE_CURRENT_BINARY_DIR}/PySide/QtGui/qabstractkineticscroller_wrapper.cpp
         )
     endif()
+else()
+  if(ENABLE_QWS)
+    set(SPECIFIC_OS_FILES
+        ${CMAKE_CURRENT_BINARY_DIR}/PySide/QtGui/qwsserver_wrapper.cpp
+        ${CMAKE_CURRENT_BINARY_DIR}/PySide/QtGui/qwsclient_wrapper.cpp
+        ${CMAKE_CURRENT_BINARY_DIR}/PySide/QtGui/qwsserver_keyboardfilter_wrapper.cpp
+    )
+  endif()
 endif()
 
 if (${QT_VERSION_MAJOR} EQUAL 4 AND ${QT_VERSION_MINOR} LESS 6)
--- /dev/null	2012-06-06 10:06:23.975309641 -0400
+++ git/PySide/QtGui/typesystem_gui_qws.xml	2012-10-10 13:03:56.242335434 -0400
@@ -0,0 +1,30 @@
+<?xml version="1.0"?>
+<!--
+    This file is part of PySide project.
+    Copyright (C) 2009-2010 Nokia Corporation and/or its subsidiary(-ies).
+    Contact: PySide team <contact@pyside.org>
+
+    This library is free software; you can redistribute it and/or
+    modify it under the terms of the GNU Lesser General Public
+    License as published by the Free Software Foundation; either
+    version 2.1 of the License, or (at your option) any later version.
+
+    This library is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+    Lesser General Public License for more details.
+
+    You should have received a copy of the GNU Lesser General Public
+    License along with this library; if not, write to the Free Software
+    Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+-->
+<typesystem package="PySide.QtGui">
+        <object-type name="QWSServer">
+                <enum-type name="ServerFlags"/>
+                <enum-type name="WindowEvent"/>
+                <enum-type name="IMMouse"/>
+                <object-type name="KeyboardFilter"/>
+        </object-type>
+        <object-type name="QWSClient"/>
+        <rejection class="QApplication" function-name="setArgs"/>
+</typesystem>

